### ISSUE TITLE
New reference panel: handle pagination correctly when mixing search-based/precise code intel

### ISSUE THREAD

**Issue by mrnugget** [sourcegraph/sourcegraph#33417]
Right now, when mixing precise & search-based is turned on, it's a bit hard to understand how pagination works and how to use it.

Why?

**If we only have precise data and NO mixing**: we load 100 precise results and if there are more results, we show the "load more references" button

**If we do NOT have precise data and NO mixing**: we load as many search-based results as we can. No pagination, since search doesn't support pagination.

**If we have precise data and mixing is ENABLED**: we load 100 precise rsults and if there are more results, we show the "load more references" button AND we load & deduplicate search-based results. 

The last one is the problem, because it results in lists like this:

![screenshot_2022-04-05_13 47 13@2x](https://user-images.githubusercontent.com/1185253/161747287-2ca98d75-b8cc-4365-9ebb-596b3e68635e.png)

Here I set the page size to 10 to replicate the behaviour. And what we end up with is 10 precise results, PLUS a bunch of search-based results (in the same repository where we have more precise results and in others), and the "load more" button is at the bottom.
 
---

We should define how pagination should work when mixing search-based & precise and we place the button in a better location.

---

**EDIT:**

Another problem @macraig I just ran into:

If we have 120 precise results and we only load the first 100, then search-based code intel might add the next 20 results. And when we then "load more" precise results, we would need to deduplicate the results too.

**Comment by sourcegraph-bot-2** [comment]
Heads up @macraig - the "team/code-intelligence" label was applied to this issue.

**Comment by mrnugget** [comment]
cc @jjinnii here too, since I could use some design help on this one.

**Comment by mrnugget** [comment]
Update: just talked to @efritz about this. This might be a non-issue, because my current implementation is wrong. We should only add search-based results once we've exhausted precise code intel.

**Comment by jjinnii** [comment]
I think the current behavior is what you mentioned in your last comment - we show search-based results only when we've gone through the precise code intel results. It'd be my recommendation to mirror the current behavior as well 😄

**Comment by mrnugget** [comment]
Okay, I fixed it in #33419. Here's a video showing how it works now (note: I set page size to `10` for this video, default is `100` which means users wouldn't have to click that many times:

https://user-images.githubusercontent.com/1185253/161938799-0ed35550-47bb-450f-8053-e48897bdff72.mp4

In text:

1. As long as we have precise results, user can paginate through them
2. Once we've reached the end of precise results (either on first request, or load-more request), we request search-based results if setting is enabled.

I also moved the "search-based"/"precise" labels to files.
